Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -1527,17 +1527,11 @@ function Set-ToolsRepo
     $newFolder = 'tools-repo'
 
     # Get all datastores
-    try
-    {
-        $datastores = Get-Datastore -ErrorAction Stop | Where-Object { $_.extensionData.Summary.Type -eq "vsan" }
-    }
-    catch
```

Comment:
Catching here is inadvisable because we can't continue. The thrown error will be reported as result of the execution.

Reply:
Is it part of the way that Run commands work that we would want to continue? If we don't find any datastores then there isn't anything left for this play to do. If we don't want an error we could wrap everything below it in an "if ($datastores) {}" so that it wouldn't error but wouldn't do anything either.